app/models/attendance.rb

```diff
@@ -17,41 +17,58 @@
 #   I'm not sure if we'll let them change it.
 #
 class Attendance < UuidApplicationRecord
-  belongs_to :child_case_cycle
+  belongs_to :child_site
+  has_one :child, through: :child_site
+  has_one :site, through: :child_site
+
+  belongs_to :child_case_cycle # TODO: really?
```

Comment:
I think so; we could do the calculation based on dates but I think getting the indexing to its child_case_cycle for free might help with performance later.